src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,115 @@
+            Scanner in = new Scanner(System.in);
+            FileCommand.loadFile(listArray);
+            FileWriter fw = new FileWriter("data/duke.txt", true);
+            boolean isLoop = true;
+            while (isLoop) {
+                String message = in.nextLine();
+                String messageLowerCase = message.toLowerCase();
+                if (messageLowerCase.equals("bye")) {
+                    exit();
+                    isLoop = false;
+                } else if (messageLowerCase.equals("list")) {
+                    try {
+                        listItems();
+                    } catch (DukeException e) {
+                        DisplayMessages.displayListMessage();
+                    }
+                } else if (messageLowerCase.contains("unmark")) {
+                    try {
+                        MarkUnmarkItem.unMarkItem(listArray, message);
+                    } catch (IndexOutOfBoundsException e) {
+                        DisplayMessages.displayUnmarkMessage(listArray);
+                    }
+                } else if (messageLowerCase.contains("mark")) {
+                    try {
+                        MarkUnmarkItem.markItem(listArray, message);
+                    } catch (IndexOutOfBoundsException e) {
+                        DisplayMessages.displayMarkMessage(listArray);
+                    }
+                } else if (messageLowerCase.contains("todo")) {
+                    try {
+                        AddTask.addTodo(listArray, message);
+                    } catch (IndexOutOfBoundsException e) {
+                        DisplayMessages.displayTodoMessage();
+                    } catch (DukeException e) {
+                        DisplayMessages.displayTodoMessage();
+                    }
+                } else if (messageLowerCase.contains("deadline")) {
+                    try {
+                        AddTask.addDeadline(listArray, message);
+                    } catch (ArrayIndexOutOfBoundsException e) {
+                        DisplayMessages.displayDeadlineMessage();
+                    } catch (DukeException e) {
+                        DisplayMessages.displayDeadlineMessage();
+                    }
+                } else if (messageLowerCase.contains("event")) {
+                    try {
+                        AddTask.addEvent(listArray, message);
+                    } catch (ArrayIndexOutOfBoundsException e) {
+                        DisplayMessages.displayEventMessage();
+                    } catch (DukeException e) {
+                        DisplayMessages.displayEventMessage();
+                    }
+                } else if (messageLowerCase.contains("delete")) {
+                    try {
+                        DeleteTask.deleteTask(listArray, message);
+                    } catch (IndexOutOfBoundsException e) {
+                        DisplayMessages.displayDeleteMessage(listArray);
+                    }
```

Comment:
You might want to consider extracting each of the `else-if` block out into a separate method for SLAP. Or alternatively, combine everything to a single `try-catch`.